File: themes/lib/targets/tmux.py
"""Tmux color theme generator."""

import os

from themes.lib.schema import ColorScheme, ThemeState

TARGET_NAME = "tmux"
ASSEMBLY = "import"
OUTPUT_PATH = "~/.config/tmux/colors.conf"
RELOAD_CMD = ["tmux", "source-file", os.path.expanduser("~/.config/tmux/colors.conf")]
COMMENT = "#"


def generate(colors: ColorScheme, state: ThemeState) -> str:
    return (
        f'set -g status-style "bg={colors.bg1},fg={colors.fg}"\n'
        f'set -g status-left "#[fg={colors.bg},bg={colors.accent},bold] #S #[bg={colors.bg1}] "\n'
        f'set -g status-right "#[fg={colors.fg}] %H:%M "\n'
        f'setw -g window-status-format " #I:#W "\n'
        f'setw -g window-status-current-format "#[fg={colors.bg},bg={colors.green},bold] #I:#W "\n'
        f'set -g pane-border-style "fg={colors.bg1}"\n'
        f'set -g pane-active-border-style "fg={colors.accent}"\n'
    )

File: desktopctl/src/theme/targets/tmux.rs
use super::{Assembly, GeneratedContent, TargetMetadata};
use crate::theme::{
    expand_user_path, run_owned_command,
    schema::{ColorScheme, ThemeState},
};

pub const METADATA: TargetMetadata = TargetMetadata {
    name: "tmux",
    assembly: Assembly::Import,
    output_path: Some("~/.config/tmux/colors.conf"),
    base_path: None,
    extra_outputs: &[],
    managed_paths: &[],
    reload_cmd: None,
    comment: Some("#"),
    sync_safe: true,
};

pub fn generate(colors: &ColorScheme, _state: &ThemeState) -> crate::Result<GeneratedContent> {
    Ok(GeneratedContent::text(format!(
        concat!(
            "set -g status-style \"bg={},fg={}\"\n",
            "set -g status-left \"#[fg={},bg={},bold] #S #[bg={}] \"\n",
            "set -g status-right \"#[fg={}] %H:%M \"\n",
            "setw -g window-status-format \" #I:#W \"\n",
            "setw -g window-status-current-format \"#[fg={},bg={},bold] #I:#W \"\n",
            "set -g pane-border-style \"fg={}\"\n",
            "set -g pane-active-border-style \"fg={}\"\n",
        ),
        colors.bg1,
        colors.fg,
        colors.bg,
        colors.accent,
        colors.bg1,
        colors.fg,
        colors.bg,
        colors.green,
        colors.bg1,
        colors.accent,
    )))
}

pub fn on_apply(_colors: &ColorScheme, _state: &ThemeState) -> crate::Result<()> {
    let output_path = expand_user_path(METADATA.output_path.expect("tmux output path"))?;
    let command = vec![
        "tmux".to_owned(),
        "source-file".to_owned(),
        output_path.display().to_string(),
    ];
    run_owned_command(&command)
}
